sdk/aqueduct/constants/enums.py

```diff
@@ -180,9 +181,15 @@ class ExecutionMode(str, Enum, metaclass=MetaEnum):
     LAZY = "lazy"
 
 
-class RuntimeType(Enum, metaclass=MetaEnum):
+class RuntimeType(str, Enum, metaclass=MetaEnum):
     AQUEDUCT = "aqueduct"
     AIRFLOW = "airflow"
     K8S = "k8s"
     LAMBDA = "lambda"
     DATABRICKS = "databricks"
+    SPARK = "spark"
+
+
+class SparkRuntimeType(str, Enum, metaclass=MetaEnum):
```

Comment:
is this the canonical way to decide if an enum is within a subset? On top of my head I would do `SPARK_RUNTIM_TYPES = [RuntimeType.SPARK, RuntimeType.DATABRICKS]` or `def requireSpark(t RuntimeType)` instead

Reply:
We should add a unit test making sure that SparkRuntimeType is always a strict subset of RuntimeType. And another one making sure that RuntimeType is a subset of ServiceType.

Reply:
@likawind The MetaEnum class allows us to use the `in` syntax - from what I remember reading it seemed like ok practice. I don't know if its canonical but it seems pythonic :P We should definitely add a comment here noting that it's a subset of RuntimeType though.